packages/protobuf/src/private/feature-set.ts

```diff
@@ -61,42 +61,35 @@ export function createFeatureResolver(
   ) {
     throw new Error("Invalid FeatureSetDefaults");
   }
-  const defaultsBinByEdition = new Map<Edition, Uint8Array>();
-  return (edition: Edition, ...rest): MergedFeatureSet => {
-    let defaultsBin: Uint8Array | undefined = defaultsBinByEdition.get(edition);
-    if (defaultsBin === undefined) {
-      if (edition < min) {
-        throw new Error(
-          `Edition ${Edition[edition]} is earlier than the minimum supported edition ${Edition[min]}`,
-        );
-      }
-      if (max < edition) {
-        throw new Error(
-          `Edition ${Edition[edition]} is later than the maximum supported edition ${Edition[max]}`,
-        );
-      }
-      let highestMatch: { e: Edition; f: FeatureSet } | undefined = undefined;
-      for (const c of compiledFeatureSetDefaults.defaults) {
-        const e = c.edition ?? 0;
-        if (e > edition) {
-          continue;
-        }
-        if (highestMatch !== undefined && highestMatch.e > e) {
-          continue;
-        }
-        highestMatch = {
-          e,
-          f: c.features ?? new FeatureSet(),
-        };
-      }
-      if (highestMatch === undefined) {
-        throw new Error(
-          `No valid default found for edition ${Edition[edition]}`,
-        );
-      }
-      defaultsBin = highestMatch.f.toBinary();
-      defaultsBinByEdition.set(edition, defaultsBin);
+  if (edition < min) {
+    throw new Error(
+      `Edition ${Edition[edition]} is earlier than the minimum supported edition ${Edition[min]}`,
+    );
+  }
```

Comment:
Throwing here right away instead of in the resolver function, the previous behavior.